### DISCUSSION TITLE
Variables, Secrets and repository overview to GitHub Teams

### DISCUSSION THREAD

**Discussion by mBlomsterberg** [community/community#112106] [Repositories]
### Select Topic Area

Product Feedback

### Body

Having the Teams in GitHub bring a useful way of assigning permissions. However, It would be more useful to get additional functionality to those Teams, such as repository lists(code responsibility/ownership) and common secrets and variables. 

This would allow us to minimize the repeatability of adding common secrets/variables of the Team and an overview of the team repositories. 

We are currently using "custom properties" to add the repository per team functionality.

**Comment by schmidkyle** [comment]
I think they should also do team runners too. I work for a company that has lots of repos and teams with in a single org. Giving out Org level access is not good practice. But allowing teams to manage runners across multiple repos makes a lot of since. The teams already have access to add runners, and secrets at each repo level, so the security issue to allow them to manage them at a team level should be minimal. 

GitHub already has much of what is needed, just in a different setup. 

Let's take what a Team Level Runner would look like. 

You can use the same function you already have with Runner Groups at the Org level. These runner groups can be setup to only allow specific repos for the runners. So let's allow teams to also act in that function. 
A team has a list of repos associated with it, you can have each team act as a single org level runner group behind the scenes. So when you add a repo to a team, it would functionally add that repo to the runner group. Then under the settings for the team you would have a section for adding a runner, much like each repo does. Everything else stays the same, there is no real reinventing the wheel here, just using what is already setup, but allowing teams to self-manage their own runners across multiple repos, without having Org level access which brings a lot more security issues. 


Now, let's take a look at what a Team level secret/variable would look like.

This too can work a lot like what you use for Org Level secrets, but just at the team level. For a team, under the setting section you would have Secrets and variables and be able to add what we will call "Team Secrets". We use the same function of what every repo is associated with, the team will show the secret in it. For limits, we can do the same as we do for org, If the repository is assigned access to more than 100 team secrets, the workflow can only use the first 100 team secrets. 
On the repo page, the order of priority would be Environment → Repository → Team → Organization. So if you had a team secret, it would override an Org secret, but not a Repo secret of the same name.

**Reply by mBlomsterberg** [reply]
Would you not just use runner groups and labels to manage the "team runners" ?

Other than that I think the secrets and variables for Teams are much more urgent.

**Comment by schmidkyle** [comment]
Runner groups can only be managed by an Enterprise admin or the Org admin. At the team level you would not have the ability to create your own runner group. 
With labels you are only able to specify who can pick up the runner, but there are several cases where you will have a very unique runner that you team will need to use, across your own repos, and in some cases you do not want other teams to use those runners as they will take up all the resources on them. Also when using something like ARC, you can only have a single label for your runner.

**Reply by mBlomsterberg** [reply]
Sure but maybe we should keep these two issues separate. My discussion concerns variables and secrets assigned to teams.

**Comment by shreeraamMNS** [comment]
+1 for team level secrets. We maintain 15 plus repositories and repeating and updating secrets in all repo is painful.
